### PR TITLE
bazel/execution/bazelapi: Plumbing for ActionResult

### PR DESCRIPTION
**Pull request by ryancouto**
This is the plumbing for the remote execution ActionResult struct, including thrift definitions, translations to domain objects, and inclusion in runner RunStatuses.

### CODE REVIEW

File: bazel/execution/bazelapi/bazel.thrift [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Structures for storing Bazel request fields in
```
Comment (ryancouto) [inline bazel/execution/bazelapi/bazel.thrift]: Note to self: Change this comment before merging

File: bazel/execution/bazelapi/definitions.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
+// Execution Request Definitions
+// Domain structures for tracking ExecuteRequest info (use cases include Scheduler JobDefs and
```
Comment (ryancouto) [inline bazel/execution/bazelapi/definitions.go]: Note to self: Change this comment before merging

File: runner/status.go [hunk 3]
```diff
@@ -94,6 +96,7 @@ func (p RunStatus) String() string {
 		s += fmt.Sprintf(" # Error: %s", p.Error)
 	}
 	s += fmt.Sprintf(" # Stdout: %s # Stderr: %s", p.StdoutRef, p.StderrRef)
+	// TODO: determine how a RunStatus with a non-nil BazelResult should look
```
Comment (dgdonut) [inline runner/status.go]: I just added this for runner.Commands (see runner/runner.go - doesn't add anything when nil), that's what the String() funcs in bazelapi are for

File: scootapi/server/job_status.go [hunk 4]
```diff
@@ -123,13 +123,14 @@ func workerRunStatusToScootRunStatus(resultsFromSaga []byte) (*scoot.RunStatus,
...
-		Error:      workerRunStatus.Error,
-		SnapshotId: workerRunStatus.SnapshotId,
+		RunId:        workerRunStatus.RunId,
+		Status:       status,
+		OutUri:       workerRunStatus.OutUri,
+		ErrUri:       workerRunStatus.ErrUri,
+		ExitCode:     workerRunStatus.ExitCode,
+		Error:        workerRunStatus.Error,
+		SnapshotId:   workerRunStatus.SnapshotId,
+		BazelResult_: workerRunStatus.BazelResult_,
```
Comment (dgdonut) [inline scootapi/server/job_status.go:133]: Why the '_'?

File: bazel/execution/bazelapi/bazel.thrift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+# tasks started from Bazel Execute Requests
+
+# See github.com/twitter/scoot/README.md for local Thrift prerequisites
+# 
+# To Generate files, run from this (github.com/twitter/scoot/bazel/execution/request/) directory:
+#     1. thrift --gen go:package_prefix=github.com/twitter/scoot/bazel/gen-go/,thrift_import=github.com/apache/thrift/lib/go/thrift bazel.thrift
```
Comment (dgdonut) [inline bazel/execution/bazelapi/bazel.thrift]: package_prefix

File: sched/sched.thrift [hunk 6]
```diff
@@ -5,9 +5,9 @@
 # See github.com/twitter/scoot/README.md for local Thrift prerequisites
 #
 # To Generate files run from this (github.com/twitter/scoot/sched/) directory:
-#     1. thrift -I ../bazel/execution/request/ --gen go:package_prefix=github.com/twitter/scoot/bazel/execution/request/gen-go/,thrift_import=github.com/apache/thrift/lib/go/thrift sched.thrift
+#     1. thrift -I ../bazel/execution/bazelapi/ --gen go:package_prefix=github.com/twitter/scoot/bazel/gen-go/,thrift_import=github.com/apache/thrift/lib/go/thrift sched.thrift
```
Comment (dgdonut) [inline sched/sched.thrift]: package_prefix

File: scootapi/scoot.thrift [hunk 7]
```diff
@@ -3,7 +3,11 @@
 # See github.com/twitter/scoot/README.md for local Thrift prerequisites
 #
 # To Generate files, run from this (github.com/twitter/scoot/scootapi/) directory:
-#     1. thrift --gen go:package_prefix=github.com/twitter/scoot/scootapi/gen-go/,package=scoot,thrift_import=github.com/apache/thrift/lib/go/thrift scoot.thrift
+#     1. thrift -I ../bazel/execution/bazelapi/ --gen go:package_prefix=github.com/twitter/scoot/bazel/gen-go/,package=scoot,thrift_import=github.com/apache/thrift/lib/go/thrift scoot.thrift
```
Comment (dgdonut) [inline scootapi/scoot.thrift]: package_prefix

ugh, maybe we should just reference the Makefile target in these comments so we have a canonical source of truth and generation method?

**Review by dgdonut — APPROVED** [review]

File: scootapi/server/job_status.go (hunk 4 above)
Comment (ryancouto) [reply scootapi/server/job_status.go:133]: Not sure - that's what the thrift generated

File: scootapi/scoot.thrift (hunk 7 above)
Comment (ryancouto) [reply scootapi/scoot.thrift]: yeah, I think that's a good call

File: runner/status.go (hunk 3 above)
Comment (ryancouto) [reply runner/status.go]: Gotcha. The duplication of fields is a little weird though. There ends up being 2 ExitCodes and a StdoutRef, StdoutDigest, and StdoutRaw (+ errs) for a RunStatus with an ActionResult

Comment (dgdonut) [reply runner/status.go]: Sure - could add an additional short/truncated string function to bazelapi and use that here

**Pull request merged**